### PR TITLE
Factory/user

### PR DESCRIPTION
**Pull request by sorja**

### CODE REVIEW

File: src/auth/authGroup.ts [hunk 1]
```diff
@@ -4,6 +4,119 @@ export interface AuthGroup {
...
+  systemAdmin: 'systemAdmin',
+  surveyAdmin: 'surveyAdmin',
+  surveyEditor: 'surveyEditor',
+  dataEditor: 'dataEditor',
+  dataCleanser: 'dataCleanser',
+  dataAnalyst: 'dataAnalyst',
+  surveyGuest: 'surveyGuest',
+}
+
+export const permissions = {
```
Comment (minotogna) [inline src/auth/authGroup.ts:21]: should these be Enums?

File: src/auth/authGroup.ts [hunk 2]
```diff
@@ -4,6 +4,95 @@ export interface AuthGroup {
   name: string
   surveyUuid?: string
   permissions: Array<Permission>
-  recordSteps: string // TODO RecordSteps
+  recordSteps: { [key: number]: string } // TODO RecordSteps
   uuid: string
 }
+
+export const groupNames = {
```
Comment (minotogna) [inline src/auth/authGroup.ts:11]: this is an ENUM. AtuhGroupName

File: src/auth/authGroup.ts [hunk 3]
```diff
@@ -4,6 +4,95 @@ export interface AuthGroup {
   name: string
```
Comment (minotogna) [inline src/auth/authGroup.ts:4]: this is of type AtuhGroupName

File: src/auth/factory.ts [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+  groupUuid: string
+}
+
+// TODO: Use Authgroup factory
+
+const defaultProps = {
+  hasProfilePicture: false,
+  status: UserStatus.INVITED,
+  title: undefined,
+  surveys: undefined,
```
Comment (minotogna) [inline src/auth/factory.ts:19]: NEVER use undefined as default value

DO NOT ADD undefined or null default values

File: src/auth/user.test.ts [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+  expect(user.name).toBe('name')
+
+  expect(user).toHaveProperty('email')
+  expect(user.email).toBe('email@email.com')
+
+  expect(user.props.title).toBeUndefined()
+
+  expect(user.status).toBe(UserStatus.INVITED)
+
+  expect(user.hasProfilePicture).toBeFalsy()
```
Comment (minotogna) [inline src/auth/user.test.ts:26]: USE MOCK OBJECTS TO VERIFY VALUES, ALWAYS!

**Review by minotogna — CHANGES_REQUESTED** [review]

**Review by SteRiccio — APPROVED** [review]

**Review by minotogna — APPROVED** [review]

**Pull request merged**